.github/workflows/main.yml

```diff
@@ -246,3 +246,101 @@ jobs:
       with:
         name: simple-system-simulation-output
         path: output_uhdm
+
+  generate-matrix-vcddiff:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.generate-matrix-vcddiff.outputs.matrix }}
+    steps:
+      - name: Checkout master
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          fetch-depth: 1
+
+      - uses: actions/setup-python@v2
+        with:
+           python-version: '3.7'
+
+      - name: Generate matrix (vcddiff)
+        id: generate-matrix-vcddiff
+        run: |
+          matrix="$(cd uhdm-integration && python list.py -d tests -s ibex synthesis OneClass hello-uvm OneThis array-copy assignment-pattern MultiplePrints OneNetInterf OneNetRange opentitan)"
+          echo "::set-output name=matrix::$matrix"
+          echo "matrix vcddiff: $matrix"
+
+  tests-vcddiff:
+    runs-on: ubuntu-latest
+    needs: [build-binaries, generate-matrix-vcddiff]
+    strategy:
+      matrix:
+        TEST_CASE_VCDDIFF: ${{ fromJson(needs.generate-matrix-vcddiff.outputs.matrix) }}
+      fail-fast:
+        false
+    env:
+      CC: gcc-9
+      CXX: g++-9
+      GIT_HTTP_LOW_SPEED_LIMIT: 1
+      GIT_HTTP_LOW_SPEED_TIME: 600
+
+    steps:
+    - name: Install dependencies
+      run: |
+        sudo apt-get update -qq
+        sudo apt install -y g++-9 build-essential cmake tclsh ant default-jre swig google-perftools libgoogle-perftools-dev python3 python3-dev uuid uuid-dev tcl-dev flex libfl-dev
+
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        fetch-depth: 1
+
+    - uses: actions/setup-python@v2
+      with:
+         python-version: '3.7'
+
+    - name: Download binaries
+      uses: actions/download-artifact@v2
+      with:
+        name: binaries
+
+    # See https://github.com/actions/upload-artifact/issues/38
+    - name: Extract
+      run: |
+        tar -xf binaries.tar
+
+    - name: Download artifact
+      run: |
+        test_name="${TEST_CASE:6}"
+        echo $test_name
+        curl -s -H "Authorization: token ${{github.token}}" -H "Accept: application/vnd.github.v3+json" "https://api.github.com/repos/${REPO}/actions/workflows/${WORKFLOW_FILE}/runs?branch=${BRANCH}&status=success" | \
+        sed -n -e 's/^.*"artifacts_url": //p' | sed 's/.\{1\}$//' | head -n1 | xargs -I '{}' curl -s -H "Authorization: token ${{github.token}}" -H "Accept: application/vnd.github.v3+json" "{}?per_page=100" | \
+        python3 .github/get_artifact_id.py ${test_name}.sv |
+        xargs -I '{}' curl -s -H "Authorization: token ${{github.token}}" -H "Accept: application/vnd.github.v3+json" -D - https://api.github.com/repos/${REPO}/actions/artifacts/{}/zip | \
+        sed -n -e 's/^.*location: //p' | sed 's/.\{1\}$//' | xargs -I"{}" curl --url "{}" --output ${test_name}.zip
+        mkdir -p uhdm-integration/build
+        unzip ${test_name}.zip -d uhdm-integration/build
```

Comment:
Could we download the artifact using the github-scripts action?

```
      - name: 'Download artifact'
        uses: actions/github-script@v3.1.0
        with:
          script: |
            var artifacts = await github.actions.listWorkflowRunArtifacts({
               owner: context.repo.owner,
               repo: context.repo.repo,
               run_id: ${{github.event.workflow_run.id }},
            });
            var matchArtifact = artifacts.data.artifacts.filter((artifact) => {
              return artifact.name == "tests_summary"
            })[0];
            var download = await github.actions.downloadArtifact({
               owner: context.repo.owner,
               repo: context.repo.repo,
               artifact_id: matchArtifact.id,
               archive_format: 'zip',
            });
            var fs = require('fs');
            fs.writeFileSync('${{github.workspace}}/tests_summary.zip', Buffer.from(download.data));
      - run: unzip tests_summary.zip -d ./out/report
```

Here is some code @lpawelcz used in another repository, maybe it could be adjusted and reused here?
All of these curl calls with multiple pipes seem kind of fragile